packages/app/src/static-props/variants/get-international-variant-chart-data.ts

```diff
@@ -25,57 +25,58 @@ export function getVariantChartData(variants: InVariants | undefined) {
     return EMPTY_VALUES;
   }
 
+  // The different historical value lists are of different lengths,
+  // so here we create a start to end range based on all of them:
+  const completeDateRange = createCompleteDateRange(variants.values);
+
   const variantsOfConcern = variants.values.filter(
-    (x) => x.last_value.is_variant_of_concern
-  );
-  const variantsOfInterest = variants.values.filter(
-    (x) => !x.last_value.is_variant_of_concern
+    (x) => last(x.values)?.is_variant_of_concern
   );
 
-  const firstVariant = variantsOfConcern.shift();
-
-  if (!isDefined(firstVariant)) {
+  if (!isDefined(completeDateRange)) {
     return EMPTY_VALUES;
   }
 
-  const values = firstVariant.values.map<VariantChartValue>((value, index) => {
-    const item = {
-      [`${firstVariant.name}_percentage`]: value.percentage,
-      [`${firstVariant.name}_occurrence`]: value.percentage,
-      date_start_unix: value.date_start_unix,
-      date_end_unix: value.date_end_unix,
-      sample_size: value.sample_size,
-    };
-
-    variantsOfConcern.forEach((variant) => {
-      item[`${variant.name}_percentage`] =
-        variant.values[index]?.percentage ?? 0;
-      item[`${variant.name}_occurrence`] =
-        variant.values[index]?.occurrence ?? 0;
-    });
-
-    const otherPercentage = variantsOfInterest.reduce(
-      (aggr, variant) => (aggr += variant.values[index]?.percentage ?? 0),
-      0
+  const values = completeDateRange.map<VariantChartValue>((partialItem) => {
+    const { item, total } = variantsOfConcern.reduce(
+      ({ item, total }, variantOfConcern) => {
+        const otherItem = variantOfConcern.values.find(
+          (x) => x.date_end_unix === partialItem.date_end_unix
+        );
+        if (otherItem) {
+          total += otherItem.percentage;
+          item[`${variantOfConcern.name}_percentage`] = otherItem.percentage;
+        }
+        return { item, total };
+      },
+      { item: partialItem, total: 0 }
     );
 
-    const otherOccurrence = variantsOfInterest.reduce(
-      (aggr, variant) => (aggr += variant.values[index]?.occurrence ?? 0),
-      0
-    );
-
-    item.other_percentage = otherPercentage;
-    item.other_occurrence = otherOccurrence;
+    item.other_percentage = Math.round((100 - total) * 100) / 100; //Round to maximum of 2 decimals
 
     return item;
   });
 
   return {
     variantChart: values,
-    dates: {
-      date_of_insertion_unix: firstVariant.last_value.date_of_insertion_unix,
-      date_start_unix: firstVariant.last_value.date_start_unix,
-      date_end_unix: firstVariant.last_value.date_end_unix,
-    },
   } as const;
 }
+
+function createCompleteDateRange(lists: InVariantsVariant[]) {
+  return lists
+    .flatMap((x) => x.values)
+    .map<VariantChartValue>((x) => ({
+      date_start_unix: x.date_start_unix,
+      date_end_unix: x.date_end_unix,
+      sample_size: x.sample_size,
+    }))
+    .filter(
+      (x, i, arr) =>
```

Comment:
nit; Unnecessary use of short-hands

```suggestion
      ({date_start_unix, date_end_unix}, index, array) =>
```

Then you can also use `x` instead of `y`